adsrefpipe/models.py

```diff
@@ -0,0 +1,256 @@
+        :param source_extension:
+        """
+        self.name = name
+        self.source_extension = source_extension
+
+    def get_name(self):
+        """
+
+        :param source_filename:
+        :return:
+        """
+        return self.name
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'name': self.name,
+            'source_extension': self.source_extension,
+        }
+
+
+class Reference(Base):
+    __tablename__ = 'reference'
+    bibcode = Column(String, primary_key=True)
+    source_filename = Column(String, primary_key=True)
+    resolved_filename = Column(String)
+    parser = Column(String, ForeignKey('parser.name'))
+
+    def __init__(self, bibcode, source_filename, resolved_filename, parser):
+        """
+
+        :param bibcode:
+        :param source_filename:
+        :param resolved_filename:
+        :param parser:
+        """
+        self.bibcode = bibcode
+        self.source_filename = source_filename
+        self.resolved_filename = resolved_filename
+        self.parser = parser
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'bibcode': self.bibcode,
+            'source_filename': self.source_filename,
+            'resolved_filename': self.resolved_filename,
+            'parser': self.parser,
+        }
+
+
+class History(Base):
+    __tablename__ = 'history'
+    __table_args__ = (ForeignKeyConstraint( ['bibcode', 'source_filename'], ['reference.bibcode', 'reference.source_filename']),)
+    id = Column(Integer, primary_key=True)
+    bibcode = Column(String)
+    source_filename = Column(String)
+    source_modified = Column(DateTime)
+    status = Column(String, ForeignKey('action.status'))
+    date = Column(DateTime, default=func.now())
+    total_ref = Column(Integer)
+
+    def __init__(self, bibcode, source_filename, source_modified, status, date, total_ref):
+        """
+
+        :param bibcode:
+        :param source_filename:
+        :param source_modified:
+        :param status:
+        :param date:
+        :param total_ref:
+        """
+        self.bibcode = bibcode
+        self.source_filename = source_filename
+        self.source_modified = source_modified
+        self.status = status
+        self.date = date
+        self.total_ref = total_ref
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'bibcode': self.bibcode,
+            'source_filename': self.source_filename,
+            'source_modified': self.source_modified,
+            'status': self.status,
+            'date': self.date,
+            'total_ref' : self.total_ref,
+        }
+
+
+class Resolved(Base):
```

Comment:
Could you find a more concrete name for this class please? `ResolvedWhat`?

Reply:
lolol. very limited vocabulary we have. so these are my explanation of what each table is all about, please feel free to give me suggestion, I did not want to say resolved_reference, since we have a table reference. I should open it up to the group for name suggestions, so https://github.com/golnazads/ADSReferencePipeline/blob/master/run.py#L291 respectively for https://github.com/golnazads/ADSReferencePipeline/blob/master/adsrefpipe/app.py#L424

Reply:
renamed to ResolvedReference